### ISSUE TITLE
Undefined symbol in shared library - SensitivityModule

### ISSUE THREAD

**Issue by lecdawson** [SuperNEMO-DBD/Falaise#72]
When running flreconstruct -i name.brio -p SensitivityModule.conf I see the following error:
 The 'SensitivityModule' library was not loaded ! DATATOOLS_SYS_NAMESPACE says: '/home/ubuntu/SensitivityModule/SensitivityModule-build/libSensitivityModule.so: undefined symbol: _ZTV8TVector3' !

With some further investigation I have found that my libSensitivityModule.so has no dependence on libPhysics.so (the library containing TVector3). I believe this is the problem, as on the UCL linux cluster the library does have a dependence on libPhysics.so. I suspect the issue is with my CMakeLists.txt.

I am running on an ubuntu virtual machine with:
- Falaise 3.1.1
- root 6.08.06
- Sensitivity Module from https://github.com/SuperNEMO-DBD/SensitivityModule

[CMakeLists.txt](https://github.com/SuperNEMO-DBD/Falaise/files/1513630/CMakeLists.txt)

**Comment by cherylepatrick** [comment]
In case this helps: this [CMakeLists.txt](https://github.com/SuperNEMO-DBD/SensitivityModule/blob/master/CMakeLists_template_linux.txt) works for me on the UCL linux cluster (but Lauren is using Ubuntu and UCL has SL6 which I think is a RedHat derivative).

@drbenmorgan do you have any ideas? We are so stuck and confused!

**Comment by drbenmorgan** [comment]
What is the output of 

```
$ ldd /path/to/libFalaise.so
$ ldd /path/to/flreconstruct
$ ldd /path/to/libSensistivityModule.so
```
? I've got a suspicion that Ubuntu's linker is using the needed/as-needed options and is thus missing a link somewhere.

**Comment by lecdawson** [comment]
$ ldd /home/ubuntu/cadfaelbrew/Cellar/falaise/3.1.1/lib/libFalaise.so
	linux-vdso.so.1 =>  (0x00007ffe26f68000)
	libBayeux.so.3 => /home/ubuntu/cadfaelbrew/lib/libBayeux.so.3 (0x00007f2aed846000)
	libboost_filesystem-mt.so.1.63.0 => /home/ubuntu/cadfaelbrew/lib/libboost_filesystem-mt.so.1.63.0 (0x00007f2aed62d000)
	libboost_serialization-mt.so.1.63.0 => /home/ubuntu/cadfaelbrew/lib/libboost_serialization-mt.so.1.63.0 (0x00007f2aed3ec000)
	libgsl.so.23 => /home/ubuntu/cadfaelbrew/lib/libgsl.so.23 (0x00007f2aecf7d000)
	libCLHEP-2.1.3.1.so => /home/ubuntu/cadfaelbrew/lib/libCLHEP-2.1.3.1.so (0x00007f2aecc33000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f2aec8b1000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f2aec5a8000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f2aec392000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f2aebfc8000)
	libboost_system-mt.so.1.63.0 => /home/ubuntu/cadfaelbrew/lib/../lib/libboost_system-mt.so.1.63.0 (0x00007f2aebdc4000)
	libboost_iostreams-mt.so.1.63.0 => /home/ubuntu/cadfaelbrew/lib/../lib/libboost_iostreams-mt.so.1.63.0 (0x00007f2aebbab000)
	libboost_program_options-mt.so.1.63.0 => /home/ubuntu/cadfaelbrew/lib/../lib/libboost_program_options-mt.so.1.63.0 (0x00007f2aeb92a000)
	libboost_regex-mt.so.1.63.0 => /home/ubuntu/cadfaelbrew/lib/../lib/libboost_regex-mt.so.1.63.0 (0x00007f2aeb613000)
	libcamp.so => /home/ubuntu/cadfaelbrew/lib/../lib/libcamp.so (0x00007f2aeb2bd000)
	libCore.so.6.08 => /home/ubuntu/cadfaelbrew/lib/root/libCore.so.6.08 (0x00007f2aeacc9000)
	libRIO.so.6.08 => /home/ubuntu/cadfaelbrew/lib/root/libRIO.so.6.08 (0x00007f2aea806000)
	libHist.so.6.08 => /home/ubuntu/cadfaelbrew/lib/root/libHist.so.6.08 (0x00007f2aea275000)
	libTree.so.6.08 => /home/ubuntu/cadfaelbrew/lib/root/libTree.so.6.08 (0x00007f2ae9f2f000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f2ae9d2b000)
	libQt5Svg.so.5 => /home/ubuntu/cadfaelbrew/opt/qt5-base/lib/libQt5Svg.so.5 (0x00007f2af2527000)
	libreadline.so.7 => /home/ubuntu/cadfaelbrew/lib/../lib/libreadline.so.7 (0x00007f2ae9ae0000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f2ae98c3000)
	libgslcblas.so.0 => /home/ubuntu/cadfaelbrew/lib/../lib/libgslcblas.so.0 (0x00007f2ae9685000)
	libQt5Widgets.so.5 => /home/ubuntu/cadfaelbrew/opt/qt5-base/lib/libQt5Widgets.so.5 (0x00007f2ae9027000)
	libQt5Gui.so.5 => /home/ubuntu/cadfaelbrew/opt/qt5-base/lib/libQt5Gui.so.5 (0x00007f2ae8a70000)
	libQt5Core.so.5 => /home/ubuntu/cadfaelbrew/opt/qt5-base/lib/libQt5Core.so.5 (0x00007f2ae853b000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f2af236e000)
	libz.so.1 => /home/ubuntu/cadfaelbrew/lib/libz.so.1 (0x00007f2ae831f000)
	libbz2.so.1.0 => /home/ubuntu/cadfaelbrew/lib/libbz2.so.1.0 (0x00007f2ae810f000)
	libicui18n.so.59 => /home/ubuntu/cadfaelbrew/lib/libicui18n.so.59 (0x00007f2ae7c8d000)
	libicuuc.so.59 => /home/ubuntu/cadfaelbrew/lib/libicuuc.so.59 (0x00007f2ae78da000)
	liblzma.so.5 => /home/ubuntu/cadfaelbrew/lib/liblzma.so.5 (0x00007f2ae76b4000)
	libThread.so.6.08 => /home/ubuntu/cadfaelbrew/Cellar/root6/6.08.06_4/lib/root/libThread.so.6.08 (0x00007f2ae7488000)
	libMatrix.so.6.08 => /home/ubuntu/cadfaelbrew/Cellar/root6/6.08.06_4/lib/root/libMatrix.so.6.08 (0x00007f2ae7136000)
	libMathCore.so.6.08 => /home/ubuntu/cadfaelbrew/Cellar/root6/6.08.06_4/lib/root/libMathCore.so.6.08 (0x00007f2ae6d45000)
	libNet.so.6.08 => /home/ubuntu/cadfaelbrew/Cellar/root6/6.08.06_4/lib/root/libNet.so.6.08 (0x00007f2ae6a7d000)
	libncursesw.so.6 => /home/ubuntu/cadfaelbrew/lib/libncursesw.so.6 (0x00007f2ae680e000)
	libGL.so.1 => /usr/lib/x86_64-linux-gnu/mesa/libGL.so.1 (0x00007f2ae659c000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f2ae6394000)
	libicudata.so.59 => /home/ubuntu/cadfaelbrew/Cellar/icu4c/59.1_1/lib/libicudata.so.59 (0x00007f2ae4881000)
	libssl.so.1.0.0 => /home/ubuntu/cadfaelbrew/lib/libssl.so.1.0.0 (0x00007f2ae460e000)
	libcrypto.so.1.0.0 => /home/ubuntu/cadfaelbrew/lib/libcrypto.so.1.0.0 (0x00007f2ae41b3000)
	libexpat.so.1 => /home/ubuntu/cadfaelbrew/lib/../lib/libexpat.so.1 (0x00007f2ae3f89000)
	libxcb-dri3.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-dri3.so.0 (0x00007f2ae3d86000)
	libxcb-present.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-present.so.0 (0x00007f2ae3b83000)
	libxcb-sync.so.1 => /usr/lib/x86_64-linux-gnu/libxcb-sync.so.1 (0x00007f2ae397c000)
	libxshmfence.so.1 => /usr/lib/x86_64-linux-gnu/libxshmfence.so.1 (0x00007f2ae3779000)
	libglapi.so.0 => /usr/lib/x86_64-linux-gnu/libglapi.so.0 (0x00007f2ae354a000)
	libXext.so.6 => /usr/lib/x86_64-linux-gnu/libXext.so.6 (0x00007f2ae3338000)
	libXdamage.so.1 => /usr/lib/x86_64-linux-gnu/libXdamage.so.1 (0x00007f2ae3135000)
	libXfixes.so.3 => /usr/lib/x86_64-linux-gnu/libXfixes.so.3 (0x00007f2ae2f2f000)
	libX11-xcb.so.1 => /usr/lib/x86_64-linux-gnu/libX11-xcb.so.1 (0x00007f2ae2d2d000)
	libX11.so.6 => /usr/lib/x86_64-linux-gnu/libX11.so.6 (0x00007f2ae29f3000)
	libxcb-glx.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-glx.so.0 (0x00007f2ae27da000)
	libxcb-dri2.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-dri2.so.0 (0x00007f2ae25d5000)
	libxcb.so.1 => /usr/lib/x86_64-linux-gnu/libxcb.so.1 (0x00007f2ae23b3000)
	libXxf86vm.so.1 => /usr/lib/x86_64-linux-gnu/libXxf86vm.so.1 (0x00007f2ae21ad000)
	libdrm.so.2 => /usr/lib/x86_64-linux-gnu/libdrm.so.2 (0x00007f2ae1f9c000)
	libbsd.so.0 => /home/ubuntu/cadfaelbrew/lib/libbsd.so.0 (0x00007f2ae1d86000)
	libXau.so.6 => /usr/lib/x86_64-linux-gnu/libXau.so.6 (0x00007f2ae1b82000)
	libXdmcp.so.6 => /usr/lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007f2ae197c000)

$ ldd ~/cadfaelbrew/bin/flreconstruct
	linux-vdso.so.1 =>  (0x00007fffaaffd000)
	libFalaise.so.3 => /home/ubuntu/cadfaelbrew/bin/../lib/libFalaise.so.3 (0x00007f9b91522000)
	libBayeux.so.3 => /home/ubuntu/cadfaelbrew/bin/../lib/libBayeux.so.3 (0x00007f9b8df29000)
	libboost_program_options-mt.so.1.63.0 => /home/ubuntu/cadfaelbrew/bin/../lib/libboost_program_options-mt.so.1.63.0 (0x00007f9b8dca8000)
	libboost_filesystem-mt.so.1.63.0 => /home/ubuntu/cadfaelbrew/bin/../lib/libboost_filesystem-mt.so.1.63.0 (0x00007f9b8da8f000)
	libboost_system-mt.so.1.63.0 => /home/ubuntu/cadfaelbrew/bin/../lib/libboost_system-mt.so.1.63.0 (0x00007f9b8d88b000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f9b8d509000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f9b8d2f3000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f9b8cf29000)
	libboost_serialization-mt.so.1.63.0 => /home/ubuntu/cadfaelbrew/bin/../lib/../lib/libboost_serialization-mt.so.1.63.0 (0x00007f9b8cce8000)
	libgsl.so.23 => /home/ubuntu/cadfaelbrew/bin/../lib/../lib/libgsl.so.23 (0x00007f9b8c879000)
	libCLHEP-2.1.3.1.so => /home/ubuntu/cadfaelbrew/bin/../lib/../lib/libCLHEP-2.1.3.1.so (0x00007f9b8c52f000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f9b8c226000)
	libboost_iostreams-mt.so.1.63.0 => /home/ubuntu/cadfaelbrew/bin/../lib/../lib/libboost_iostreams-mt.so.1.63.0 (0x00007f9b8c00d000)
	libboost_regex-mt.so.1.63.0 => /home/ubuntu/cadfaelbrew/bin/../lib/../lib/libboost_regex-mt.so.1.63.0 (0x00007f9b8bcf6000)
	libcamp.so => /home/ubuntu/cadfaelbrew/bin/../lib/../lib/libcamp.so (0x00007f9b8b9a0000)
	libCore.so.6.08 => /home/ubuntu/cadfaelbrew/lib/root/libCore.so.6.08 (0x00007f9b8b3ac000)
	libRIO.so.6.08 => /home/ubuntu/cadfaelbrew/lib/root/libRIO.so.6.08 (0x00007f9b8aee9000)
	libHist.so.6.08 => /home/ubuntu/cadfaelbrew/lib/root/libHist.so.6.08 (0x00007f9b8a958000)
	libTree.so.6.08 => /home/ubuntu/cadfaelbrew/lib/root/libTree.so.6.08 (0x00007f9b8a612000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f9b8a40e000)
	libQt5Svg.so.5 => /home/ubuntu/cadfaelbrew/opt/qt5-base/lib/libQt5Svg.so.5 (0x00007f9b92c09000)
	libreadline.so.7 => /home/ubuntu/cadfaelbrew/bin/../lib/../lib/libreadline.so.7 (0x00007f9b8a1c3000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f9b89fa6000)
	libgslcblas.so.0 => /home/ubuntu/cadfaelbrew/bin/../lib/../lib/libgslcblas.so.0 (0x00007f9b89d68000)
	libQt5Widgets.so.5 => /home/ubuntu/cadfaelbrew/opt/qt5-base/lib/libQt5Widgets.so.5 (0x00007f9b8970a000)
	libQt5Gui.so.5 => /home/ubuntu/cadfaelbrew/opt/qt5-base/lib/libQt5Gui.so.5 (0x00007f9b89153000)
	libQt5Core.so.5 => /home/ubuntu/cadfaelbrew/opt/qt5-base/lib/libQt5Core.so.5 (0x00007f9b88c1e000)
	/home/ubuntu/cadfaelbrew/lib/ld.so => /lib64/ld-linux-x86-64.so.2 (0x00007f9b92a51000)
	libz.so.1 => /home/ubuntu/cadfaelbrew/lib/libz.so.1 (0x00007f9b88a02000)
	libbz2.so.1.0 => /home/ubuntu/cadfaelbrew/lib/libbz2.so.1.0 (0x00007f9b887f2000)
	libicui18n.so.59 => /home/ubuntu/cadfaelbrew/lib/libicui18n.so.59 (0x00007f9b88370000)
	libicuuc.so.59 => /home/ubuntu/cadfaelbrew/lib/libicuuc.so.59 (0x00007f9b87fbd000)
	liblzma.so.5 => /home/ubuntu/cadfaelbrew/lib/liblzma.so.5 (0x00007f9b87d97000)
	libThread.so.6.08 => /home/ubuntu/cadfaelbrew/Cellar/root6/6.08.06_4/lib/root/libThread.so.6.08 (0x00007f9b87b6b000)
	libMatrix.so.6.08 => /home/ubuntu/cadfaelbrew/Cellar/root6/6.08.06_4/lib/root/libMatrix.so.6.08 (0x00007f9b87819000)
	libMathCore.so.6.08 => /home/ubuntu/cadfaelbrew/Cellar/root6/6.08.06_4/lib/root/libMathCore.so.6.08 (0x00007f9b87428000)
	libNet.so.6.08 => /home/ubuntu/cadfaelbrew/Cellar/root6/6.08.06_4/lib/root/libNet.so.6.08 (0x00007f9b87160000)
	libncursesw.so.6 => /home/ubuntu/cadfaelbrew/lib/libncursesw.so.6 (0x00007f9b86ef1000)
	libGL.so.1 => /usr/lib/x86_64-linux-gnu/mesa/libGL.so.1 (0x00007f9b86c7f000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f9b86a77000)
	libicudata.so.59 => /home/ubuntu/cadfaelbrew/Cellar/icu4c/59.1_1/lib/libicudata.so.59 (0x00007f9b84f64000)
	libssl.so.1.0.0 => /home/ubuntu/cadfaelbrew/lib/libssl.so.1.0.0 (0x00007f9b84cf1000)
	libcrypto.so.1.0.0 => /home/ubuntu/cadfaelbrew/lib/libcrypto.so.1.0.0 (0x00007f9b84896000)
	libexpat.so.1 => /home/ubuntu/cadfaelbrew/bin/../lib/../lib/libexpat.so.1 (0x00007f9b8466c000)
	libxcb-dri3.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-dri3.so.0 (0x00007f9b84469000)
	libxcb-present.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-present.so.0 (0x00007f9b84266000)
	libxcb-sync.so.1 => /usr/lib/x86_64-linux-gnu/libxcb-sync.so.1 (0x00007f9b8405f000)
	libxshmfence.so.1 => /usr/lib/x86_64-linux-gnu/libxshmfence.so.1 (0x00007f9b83e5c000)
	libglapi.so.0 => /usr/lib/x86_64-linux-gnu/libglapi.so.0 (0x00007f9b83c2d000)
	libXext.so.6 => /usr/lib/x86_64-linux-gnu/libXext.so.6 (0x00007f9b83a1b000)
	libXdamage.so.1 => /usr/lib/x86_64-linux-gnu/libXdamage.so.1 (0x00007f9b83818000)
	libXfixes.so.3 => /usr/lib/x86_64-linux-gnu/libXfixes.so.3 (0x00007f9b83612000)
	libX11-xcb.so.1 => /usr/lib/x86_64-linux-gnu/libX11-xcb.so.1 (0x00007f9b83410000)
	libX11.so.6 => /usr/lib/x86_64-linux-gnu/libX11.so.6 (0x00007f9b830d6000)
	libxcb-glx.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-glx.so.0 (0x00007f9b82ebd000)
	libxcb-dri2.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-dri2.so.0 (0x00007f9b82cb8000)
	libxcb.so.1 => /usr/lib/x86_64-linux-gnu/libxcb.so.1 (0x00007f9b82a96000)
	libXxf86vm.so.1 => /usr/lib/x86_64-linux-gnu/libXxf86vm.so.1 (0x00007f9b82890000)
	libdrm.so.2 => /usr/lib/x86_64-linux-gnu/libdrm.so.2 (0x00007f9b8267f000)
	libbsd.so.0 => /home/ubuntu/cadfaelbrew/lib/libbsd.so.0 (0x00007f9b82469000)
	libXau.so.6 => /usr/lib/x86_64-linux-gnu/libXau.so.6 (0x00007f9b82265000)
	libXdmcp.so.6 => /usr/lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007f9b8205f000)

$ldd libSensitivityModule.so 
	linux-vdso.so.1 =>  (0x00007ffd28bb9000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f4629673000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f462936a000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f4629154000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f4628d8a000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f4629c64000)

**Comment by lecdawson** [comment]
Hi Ben,

I have now created a SL6 virtual box and have a fully working system. Therefore, for myself there is no need for this fix. However, this may be an issue that other people come across if using Ubuntu.

**Comment by drbenmorgan** [comment]
See [Issue #8](https://github.com/SuperNEMO-DBD/SensitivityModule/issues/8) in SensitivityModule for a resolution. I'll add documentation on linking modules with libraries outside the core supplied by `flreconstruct`.

**Comment by drbenmorgan** [comment]
Closing this as "fixed" by #78
